src/resources/resource_device.py

```diff
@@ -1,40 +1,34 @@
-from flask_restful import Resource, reqparse, abort, marshal_with
+from flask_restful import Resource, abort
+from flask_restful.reqparse import request
 
 from src.models.device.model_device import DeviceModel
-from src.resources.rest_schema.schema_device import device_all_attributes, device_all_fields
+from src.resources.rest_schema.schema_device import device_all_fields, device_all_fields_with_children
+from src.resources.utils import model_marshaller_with_children
 
 
-class DeviceResource(Resource):
-    parser = reqparse.RequestParser()
-    for attr in device_all_attributes:
-        parser.add_argument(attr,
-                            type=device_all_attributes[attr]['type'],
-                            required=device_all_attributes[attr].get('required', False),
-                            help=device_all_attributes[attr].get('help', None),
-                            store_missing=False)
+def device_marshaller(data: any, args: dict):
+    return model_marshaller_with_children(data, args, device_all_fields, device_all_fields_with_children)
```

Comment:
We can use decorator for such cases, but fine for now.

Reply:
nah I didn't use the decorator as I return 2 different marshallings depending on the query string